sklearnex/neighbors/knn_classification.py

```diff
@@ -15,4 +15,164 @@
+                     **kwargs):
+            super().__init__(
+                n_neighbors=n_neighbors,
+                algorithm=algorithm,
+                leaf_size=leaf_size, metric=metric, p=p,
+                metric_params=metric_params,
+                n_jobs=n_jobs, **kwargs)
+            self.weights = \
+                weights if LooseVersion(sklearn_version) >= LooseVersion("1.0") else _check_weights(weights)
+elif LooseVersion(sklearn_version) >= LooseVersion("0.22"):
+    from sklearn.neighbors._base import SupervisedIntegerMixin as \
+        BaseSupervisedIntegerMixin
+
+    class KNeighborsClassifier_(onedal_KNeighborsClassifier, onedal_KNeighborsMixin,
+                                BaseSupervisedIntegerMixin, BaseClassifierMixin):
+        @_deprecate_positional_args
+        def __init__(self, n_neighbors=5, *,
+                     weights='uniform', algorithm='auto', leaf_size=30,
+                     p=2, metric='minkowski', metric_params=None, n_jobs=None,
+                     **kwargs):
+            super().__init__(
+                n_neighbors=n_neighbors,
+                algorithm=algorithm,
+                leaf_size=leaf_size, metric=metric, p=p,
+                metric_params=metric_params,
+                n_jobs=n_jobs, **kwargs)
+            self.weights = _check_weights(weights)
+else:
+    from sklearn.neighbors.base import SupervisedIntegerMixin as \
+        BaseSupervisedIntegerMixin
+
+    class KNeighborsClassifier_(onedal_KNeighborsClassifier, onedal_KNeighborsMixin,
+                                BaseSupervisedIntegerMixin, BaseClassifierMixin):
+        @_deprecate_positional_args
+        def __init__(self, n_neighbors=5, *,
+                     weights='uniform', algorithm='auto', leaf_size=30,
+                     p=2, metric='minkowski', metric_params=None, n_jobs=None,
+                     **kwargs):
+            super().__init__(
+                n_neighbors=n_neighbors,
+                algorithm=algorithm,
+                leaf_size=leaf_size, metric=metric, p=p,
+                metric_params=metric_params,
+                n_jobs=n_jobs, **kwargs)
+            self.weights = _check_weights(weights)
+
+class KNeighborsClassifier(KNeighborsClassifier_):
+    @_deprecate_positional_args
+    def __init__(self, n_neighbors=5, *,
+                 weights='uniform', algorithm='auto', leaf_size=30,
+                 p=2, metric='minkowski', metric_params=None, n_jobs=None,
+                 **kwargs):
+        super().__init__(
+            n_neighbors=n_neighbors,
+            weights=weights,
+            algorithm=algorithm,
+            leaf_size=leaf_size, metric=metric, p=p,
+            metric_params=metric_params,
+            n_jobs=n_jobs, **kwargs)
+
+    def fit(self, X, y):
+        X_incorrect_type = isinstance(
+            X, (KDTree, BallTree, onedal_KNeighborsClassifier, BaseNeighborsBase))

```

Comment:
I did not understand this condition. why do you compare with `onedal_KNeighborsClassifier`?